sound/soc/sof/intel/hda-stream.c

```diff
@@ -446,6 +447,9 @@ int hda_dsp_stream_hw_params(struct snd_sof_dev *sdev,
 		return -ENODEV;
 	}
 
+	/* the probe routine already checks if the descriptor is valid */
+	chip = get_chip_info(sdev->pdata);
```

Comment:
I would move this as
```c
const struct sof_intel_dsp_desc *chip = get_chip_info(sdev->pdata);
```
I don't think the comment helps in any way and worth the additional lines.

Reply:
I initially wanted to initialize the pointer only after the error check, but you're right it's probably overkill. will change.